test/components/views/settings/tabs/room/NotificationSettingsTab-test.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { fireEvent, render, RenderResult } from "@testing-library/react";
+import { MatrixClient } from "matrix-js-sdk/src/client";
+
+import NotificationSettingsTab from "../../../../../../src/components/views/settings/tabs/room/NotificationSettingsTab";
+import { mkStubRoom, stubClient } from "../../../../../test-utils";
+import { MatrixClientPeg } from "../../../../../../src/MatrixClientPeg";
+
+describe("NotificatinSettingsTab", () => {
+    const roomId = "!room:example.com";
+    let cli: MatrixClient;
+
+    const renderTab = (): RenderResult => {
+        return render(<NotificationSettingsTab roomId={roomId} closeSettingsFn={() => { }} />);
+    };
+
+    beforeEach(() => {
+        stubClient();
+        cli = MatrixClientPeg.get();
+        mkStubRoom(roomId, "test room", cli);
+
+        NotificationSettingsTab.contextType = React.createContext(cli);
+    });
+
+    it("should prevent »Settings« link click from bubbling up to radio buttons", () => {
+        const tab = renderTab();
+        const event = new MouseEvent("click", { bubbles: true });
+        Object.assign(event, { preventDefault: jest.fn() });
+
+        const settingsLink = tab.container.querySelector("div.mx_AccessibleButton");
+        if (!settingsLink) throw new Error("settings link does not exist.");
+
+        fireEvent(settingsLink, event);
+        expect(event.preventDefault).toHaveBeenCalledTimes(1);
```

Comment:
It bother me that we check here the call preventDefault for the test verification. In my opinion, we should test that we do not modify the notification settings when we click on the settings link. We test here the method to fix instead of the bug himself

Instead of checking the preventDefault, maybe we can check if the this.roomProps.notificationVolume attribute has change after the click or/and the `checked` attribute of the radio button .

Reply:
I totally agree. Great hint with the `roomProps`!